### PR TITLE
blockdev_snapshot_chains:support raw

### PR DESCRIPTION
**Pull request by aliang123**
Support for case "blockdev_snapshot_chains"

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2023659

### CODE REVIEW

**Comment by aliang123** [comment]
base on:https://github.com/avocado-framework/avocado-vt/pull/3279

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u6.product_rhel.raw.virtio_blk.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_chains: PASS (78.83 s)

File: qemu/tests/blockdev_snapshot_chains.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def prepare_snapshot_file(self):
             self.params["image_name_%s" % snapshot_tag] = snapshot_tag
             snapshot_format = params.get("snapshot_format", "qcow2")
             params["image_format_%s" % snapshot_tag] = snapshot_format
+            self.params["image_format_%s" % snapshot_tag] = snapshot_format
             image = sp_admin.volume_define_by_params(snapshot_tag, params)
             image.hotplug(self.main_vm)
```
Comment (ngu-niny) [inline qemu/tests/blockdev_snapshot_chains.py:32]: In 'verify_snapshot()', we need to add following lines 59/60 to let the following rebase work:

 55     def verify_snapshot(self):
 56         if self.main_vm.is_alive():
 57             self.main_vm.destroy()
 58         base_tag = self.base_tag
 59        ** image_format = self.params.get("image_format", "qcow2")
 60        ** self.params["image_format_%s" % base_tag] = image_format

**Comment by zhencliu** [comment]
> base on:[avocado-framework/avocado-vt#3279](https://github.com/avocado-framework/avocado-vt/pull/3279)

hi @aliang123 , I am afraid we cannot change the VT code for the purpose that we wanna pass some cases, for the snapshot cases, in fact, we 'hacked' the rebase function to bypass some checkings to do the image rebase, this can always work fine when:
we set the format of data and snapshot images explicitly, e.g.  image_format_data = raw, image_format_sn1 = qcow2, or
we always use qcow2(imageformat=qcow2) format for the images

But this can fail when:
We don't set image_format_data in cfg, and we use raw(imageformat=raw) format for the images.

To avoid the failure, we should specify the image_format explicitly for both snapshot and data images, I think we can do it in  __init__ in blockdev_snapshot_base:
self.params.setdefault('image_format_%s' % self.base_tag,
                               self.params.get('image_format', 'qcow2'))
        self.params.setdefault('image_format_%s' % self.snapshot_tag,
                               self.params.get('snapshot_format', 'qcow2'))

Besides, for some cases with more than one base image or snapshot image, I am afraid we have to update the py or cfg file with image_format(e.g. blockdev_snapshot_chain case with 64 snapshot images).

But I don't think there are many cases to be modified, because we often set image_format_sn/image_format_data explicitly in cfg file.
# grep rebase blockdev_* | cut -d: -f1 | uniq | wc -l
13
(Total 13 blockdev_* cases call rebase function)

**Comment by zhencliu** [comment]
I ran all the 13 cases, and found that blockdev_snapshot_multi_disks needs to be updated too, just fyi @aliang123

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u0.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_chains.q35: PASS (87.82 s)

**Comment by zhencliu** [comment]
LGTM

**Comment by ngu-niny** [comment]
(2/2) Host_RHEL.m8.u6.product_rhel.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.blockdev_snapshot_chains: PASS (347.50 s)

Acked-by: Nini Gu <ngu@redhat.com

**Pull request merged**